LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs

```diff
@@ -2183,6 +2185,64 @@ private static void GetFintekConfiguration(ISuperIO superIO, Manufacturer manufa
     {
         switch (manufacturer)
         {
+            case Manufacturer.ASRock:
+                switch (model)
+                {
+                    case Model.IMB_1222:
+                        v.Add(new Voltage("+3V", 0, 150, 150));
+                        v.Add(new Voltage("+3VSB", 5, 150, 150));
+                        v.Add(new Voltage("VCore", 1));
+                        v.Add(new Voltage("VCCM", 2, 150, 150));
+                        v.Add(new Voltage("Voltage #4", 3, true));
+                        v.Add(new Voltage("Voltage #5", 4, true));
+                        v.Add(new Voltage("VBAT", 6, 150, 150));
+                        v.Add(new Voltage("Voltage #8", 7, true));
+                        v.Add(new Voltage("Voltage #9", 8, true));
+                        for (int i = 0; i < superIO.Temperatures.Length; i++)
+                        {
+                            if (i == 0)
+                            {
+                                t.Add(new Temperature("CPU Temperature", i));
+                            }
+                            else if (i == 1)
+                            {
+                                t.Add(new Temperature("M/B Temperature", i));
+                            }
+                            else
+                            {
+                                t.Add(new Temperature("Temperature #" + (i + 1), i));
+                            }
+                        }
+                        for (int i = 0; i < superIO.Fans.Length; i++)
+                            f.Add(new Fan("Fan #" + (i + 1), i));
+                        for (int i = 0; i < superIO.Controls.Length; i++)
+                            c.Add(new Control("Fan Control #" + (i + 1), i));
+                        break;
```

Comment:
Some pointers I wanted to give out (I'm not a maintainer, but I do go out of my way to add motherboard-specific sensor support when possible):

# Naming schemes

You might get some requested changes on the naming scheme. Typically to avoid abbreviated names or duplicate namings.

If you want a good reference, I have a couple PRs for just that: #1364, #1360, and #1343

Some suggested naming changes:

`+3V` -> `+3.3V` (Typically the more accurate name)
`+3VSB` -> `+3V Standby`
`VBAT` -> `CMOS Battery`
`CPU Temperature` -> A more reflective name of the actual CPU portion that's being checked (`CPU Core`, `CPU Package`, or just generic `CPU` if unsure).
`M/B Temperature` -> `Motherboard`

# Temperatures
Some temperature sensors may be auxiliary sensors. From my experience with ASRock boards, it typically goes as the following:

Index 0: CPU (PECI) <- usually the proper index
Index 1: Either Motherboard or another potential CPU temperature, though commonly inaccurate
Index 2: Usually the Motherboard.

The rest of the sensors onward would be auxiliary sensors

Auxiliary sensors are typically "noise" sensors and are not worth showing.

Since you're adding a specific board. I recommend doing a bit of due diligence to check what sensors "officially" show up. Primarily in the UEFI and/or any software provided from ASRock that would show these temperatures.

Remember that even though the SuperIO chip may show up to a certain limit, the motherboard may not actually include that said amount (example the SuperIO chip can support 7 fans, but the motherboard may only use 3)

# Fans
Only thing I have to say is that I'd do a bit of a deeper look to properly label the type of fan (On which index is perhaps the CPU Fan, Chassis Fan, and Sata Power Fan (if I'm reading the motherboard labels correctly, it's a bit confusing with their label placement) set at?)

![image](https://github.com/user-attachments/assets/0316cfea-b6bc-48bc-bd7b-0c3e3a7102f1)

Also, if any of the fans aren't able to be controlled ("Power"/PSU fans and CPU Optional fans on some vendors may not have a control), I'd also adding those as a control if you ever encounter that.